PersonaFlow/agentstack
# PF-221: Implement ingestion status API

Implements a new API at /rag/ingest/{task_id}/progress

There is an issue where the backend streams all of the events in when the ingestion process has finished that still needs to be resolved, but its enough for the UI to implement.

Test steps:

1. in postman, call the /ingest api with a couple of files using the semantic chunking method so it takes some time to run. This will return:
{
    "task_id": "994bc452-09aa-4690-8673-dc9282c20d91",
    "status": "started"
}

2. Copy the task_id and open a terminal and paste in the following: `curl  http://localhost:9000/api/v1/rag/ingest/{task_id}/progress`

It will take a few seconds due to the issue of it not streaming the events until after flow completes, but you should see:
```
event: metadata
data: {"ingestion_id":"c96981c9-5d5b-4afb-a1e5-b41b9d912d74"}

event: data
data: {"progress":"Retrieving file 1/2"}

event: data
data: {"progress":"Retrieved file 1/2: llm-as-judge.pdf"}

event: data
data: {"progress":"Retrieving file 2/2"}

event: data
data: {"progress":"Retrieved file 2/2: small.json"}

event: data
data: {"progress":"Generating chunks"}

event: data
data: {"progress":"Processing file 1/2: llm-as-judge.pdf"}

event: data
data: {"progress":"Processing file 2/2: small.json"}

event: data
data: {"progress":"Generated 421 chunks"}

(...)

event: data
data: {"status":"completed"}

event: end

```

Note: make sure to use the semantic splitting method which takes a little longer to run, otherwise by the time you make the second request the process may be finished.
